### PR TITLE
Remove Task/Coeval instances

### PR DESCRIPTION
**Pull request by alexandru**
`Local` should have side-effectful behavior, should not suspend execution, otherwise its usage is misleading.

I've added an import for synchronous behavior:

```scala
import monix.execution.misc.CanBindLocals.Implicits.synchronousAsDefault
```

What this does is to import in scope an implicit for all unknown `CanBindLocals[A]` types, but that does NOT override the implicits that are already in scope. This is important because we don't want to override the `Future` instance via an import like this.

Towards this purpose I've published a new micro library that introduces the `Not` data type, for providing evidence that an implicit does not exist for a certain type:
https://github.com/monix/implicitbox

This is a very light dependency and introducing it in Monix felt wrong.

### CODE REVIEW

**Comment by oleg-py** [comment]
Do we need to remove them? Can't we eagerly capture locals, and suspend the rest?

**Comment by alexandru** [comment]
Don’t know, maybe.

**Comment by alexandru** [comment]
@Avasil @oleg-py I've updated this branch to introduce an `CanBindLocals.Implicits.synchronousAsDefault` symbol that can be imported to provide synchronous behavior for all data types, but that does not override the implicits that are already in scope.

I've also updated the description of the PR.

File: monix-execution/shared/src/main/scala/monix/execution/misc/CanBindLocals.scala [hunk 1]
```diff
@@ -19,84 +19,133 @@ package monix.execution.misc
...
     */
   implicit def completableFuture[R]: CanBindLocals[CompletableFuture[R]] =
     CompletableFutureInstance.asInstanceOf[CanBindLocals[CompletableFuture[R]]]
 
+  object Implicits {
+    /**
+      * Implicit instance for all things synchronous.
+      *
+      * Needs to be imported explicitly in scope. Will NOT override
+      * other `CanBuildFrom` implicits that are already visible.
```
Comment (Avasil) [inline monix-execution/shared/src/main/scala/monix/execution/misc/CanBindLocals.scala]:
```suggestion
      * other `CanBindLocals` implicits that are already visible.
```

**Review by Avasil — APPROVED** [review]

**Comment by oleg-py** [comment]
Looks neat, let's go for it.

**Pull request merged**